base_layer/wallet/src/output_manager_service/service.rs

```diff
@@ -731,11 +774,17 @@ where
 
         let (spending_key, script_private_key) = self.get_spend_and_script_keys().await?;
 
+        let mut updated_features = single_round_sender_data.features.clone();
```

Comment:
Is this correct? I don't think the sender's features should be updated

Reply:
These are the features suggested by the sender for the receiver's output, then the receiver has to update the recovery byte.